File: src/main/java/commands/slashadapters/adapters/DeepDreamAdapter.java
package commands.slashadapters.adapters;

import commands.runnables.aitoyscategory.DeepDreamCommand;
import commands.slashadapters.Slash;
import commands.slashadapters.SlashMeta;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;

@Slash(command = DeepDreamCommand.class)
public class DeepDreamAdapter extends AIAdapterAbstract {

    @Override
    public SlashMeta process(SlashCommandEvent event) {
        return new SlashMeta(DeepDreamCommand.class, collectArgs(event));
    }

}

File: src/main/java/commands/slashadapters/adapters/DeepDreamAdapter.kt
package commands.slashadapters.adapters

import commands.runnables.aitoyscategory.DeepDreamCommand
import commands.slashadapters.Slash
import commands.slashadapters.SlashMeta
import mysql.hibernate.entity.GuildEntity
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent

@Slash(command = DeepDreamCommand::class)
class DeepDreamAdapter : AIAdapterAbstract() {

    override fun process(event: SlashCommandInteractionEvent, guildEntity: GuildEntity): SlashMeta {
        return SlashMeta(DeepDreamCommand::class.java, collectArgs(event))
    }

}